L3S-03.md

```diff
@@ -0,0 +1,44 @@
+# L3S-03 - Unit Testing
+
+- All pure functions require unit tests.
+- All unit tests should follow the Arrange, Act, Assert (AAA) pattern.
+- NodeJS projects should maintain tests in a folder `test` in the top-level of the project.
+- `mocha`+`chai` or `jest` are good unit testing framework choices for NodeJS projects.
+- NodeJS unit tests using `mocha` or `jest` should conform to the following format:
+  1. The `describe` block is the message signature of the tested function.
+  2. There is a **minimum** of one `it` block per return path for the function.
+  3. Each `expect` assertion must include a debugging message logging the value tested to assist ease of debugging.
+
+Here is an example of a conforming unit test...
+
+```
+/**
+ * Dummy function to return 'foo' or 'bar'
+ * @param baz A flag that that if true, returns 'foo'. Else, returns 'bar'
```

Comment:
* @param baz A flag that if true, returns 'foo'. Else, returns 'bar'

Reply:
```suggestion
 * @param baz A flag that if true, returns 'foo'. Else, returns 'bar'
```